src/envoy/http/authn/authenticator_base_test.cc

```diff
@@ -123,6 +141,37 @@ TEST_F(AuthenticatorBaseTest,
 }
 
 // TODO: more tests for Jwt.
+TEST_F(AuthenticatorBaseTest, ValidateJwtWithNoJwtInHeader) {
+  iaapi::Jwt jwt;
+  authenticator_.validateJwt(jwt, [](const Payload* payload, bool success) {
+    // When there is no JWT in the HTTP header, validateJwt() should return
+    // nullptr and failure.
+    EXPECT_TRUE(payload == nullptr);
+    EXPECT_FALSE(success);
+  });
+}
+
+TEST_F(AuthenticatorBaseTest, ValidateJwtWithJwtInHeader) {
+  iaapi::Jwt jwt;
+  Http::TestHeaderMapImpl request_headers_with_jwt{
+      {kSecIstioAuthUserInfoHeaderKey, kSecIstioAuthUserinfoHeaderValue}};
+  FilterContext filter_context{&request_headers_with_jwt, &connection_};
+  MockAuthenticatorBase authenticator{&filter_context};
+
+  authenticator.validateJwt(jwt, [](const Payload* payload, bool success) {
+    // When there is a verified JWT in the HTTP header, validateJwt()
+    // should return non-nullptr and success.
+    EXPECT_TRUE(payload != nullptr);
+    EXPECT_TRUE(success);
+    EXPECT_STREQ(kJwtUserExpected.c_str(), payload->jwt().user().c_str());
```

Comment:
Use  EXPECT_TRUE with TestUtility::protoEqual to test all data in proto.

Reply:
Done.  TestUtility::protoEqual() uses SerializeAsString() and the output is non-deterministic.  Thus, MessageDifferencer::Equals() is used, in which "Equality is defined as  all fields within the two messages being set to the same value. Primitive fields and strings are compared by value while embedded messages/groups are compared as if via a recursive call. "